xdsl/irdl/irdl.py

```diff
@@ -1472,6 +1472,14 @@ def wrong_field_exception(field_name: str) -> PyRDLOpDefinitionError:
             annotations = parent_cls.__annotations__
 
             for field_name in annotations:
+                if field_name.isupper() and (
+                    get_origin(annotations[field_name]) is ClassVar
+                    or (
+                        isinstance(annotations[field_name], str)
+                        and annotations[field_name].startswith("ClassVar")
+                    )
+                ):
+                    continue
```

Comment:
Seeing as we have this in two places now, could we factor this out to make it easier to see that the two checks are identical? And maybe also make this code denser again?

Reply:
That sounds very reasonable, have factored out the check as suggested.